python/pyarrow/_dataset.pyx

```diff
@@ -2969,6 +2982,8 @@ def _filesystemdataset_write(
     FileSystem filesystem not None,
     Partitioning partitioning not None,
     FileWriteOptions file_options not None,
+    bint use_threads,
+    bint use_async,
```

Comment:
This doesn't seem used here? Though I guess even before use_threads was ignored.

Reply:
I guess we'd have to do something like embed ScanOptions into FileSystemDatasetWriteOptions if we want to make these useful, and you already made a followup.

Reply:
Nope, you're right initially.  If the function is already taking in a scanner there is no need for these properties.  They should be specified when building the scanner.